integration/harness.go

```diff
@@ -102,14 +101,11 @@ func (h *Harness) RESTListen() string {
 
 // connectClient attempts to establish a gRPC Client connection
 // to the provided target.
-func connectClient(target string) (*grpc.ClientConn, error) {
-	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+func connectClient(ctx context.Context, target string) (*grpc.ClientConn, error) {
 	opts := []grpc.DialOption{
 		grpc.WithTransportCredentials(insecure.NewCredentials()),
 		grpc.WithBlock(),
```

Comment:
Consider adding a timeout. We cannot tell here if the parent `ctx` has any timeout on it.
```suggestion
		grpc.WithTransportCredentials(insecure.NewCredentials()),
		grpc.WithBlock(),
		grpc.WithTimeout(time.Second * 10),
```

Reply:
This doesn't really do anything: `grpc.WithTimeout` forks the passed in `ctx` by creating a new `context` with the given timeout: https://github.com/grpc/grpc-go/blob/v1.50.1/clientconn.go#L217-L221

If the context already has a timeout this only makes sure that the timeout isn't more than 10 seconds.

Reply:
> EDIT2: `grpc.WithTimeout` is also marked as deprecated with the hint of using `grpc.DialContext` and a context instead.

Oh, too bad :( Then I would create a child `Context` with timeout and pass it to `grpc.DialContext(dialCtx)` to make sure there is a reasonable timeout for the connection.